File: alabala/xml-ws/xml-validator/web-client/xml-validator-client.cs
/*
   Un program-client C# utilizat la validarea documentelor XML, 
   invocand metode ale unui serviciu Web via un proxy generat automat
   
   Autor: Sabin-Corneliu Buraga - http://profs.info.uaic.ro/~busaco/
   Ultima actualizare: 24 noiembrie 2007
*/
using System;

class XMLValidatorClient {
   public static void Main(string[] args) {
     
     if (args.Length == 0) {
     	  Console.WriteLine("Client de validare via un serviciu Web a unui document XML stocat pe server.");
     	  Console.WriteLine("Sintaxa: xml-validator-client <document.xml>");
     	  return;
     }
     
     try {
     	Console.Write("Documentul '{0}' este... ", args[0]);
        // cream o instanta a clasei corespunzatoare serviciului Web     
        XMLValidator validator = new XMLValidator();
     
        // invocam metoda de verificare a existentei fisierului
        if (validator.CheckIfExists(args[0])) {
          // exista, deci putem realiza validarea
          string valid = validator.Validate(args[0]);
          // afisam ce a returnat metoda Validate() invocata 
          Console.WriteLine(valid + ".");
        }   
        else      
          Console.WriteLine("inaccesibil (nu exista pe server?).");
     } // final de 'try'
     catch (Exception e) {
        // a aparut o exceptie
      	Console.WriteLine("\nExceptie: " + e.Message);
      	return;
     }	     
  }  
} // final de clasa

File: alabala/xml-ws/xml-validator/web-client/xml-validator-client.php
<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="ro" xml:lang="ro">
<head>
<title>Un client SOAP</title>
<style type="text/css">
.cod { overflow: scroll; white-space: normal; }
</style>
</head>
<body>
<?php
/* Un program-client PHP5 utilizat la validarea documentelor XML, 
   invocand metode ale unui serviciu Web implementat sub .NET
   si folosind extensia SOAP

   Autor: Sabin-Corneliu Buraga - http://profs.info.uaic.ro/~busaco/
   Ultima actualizare: 24 noiembrie 2007
*/
$doc = $_REQUEST['xml'];
if (!$doc) 
  die ("Furnizati numele documentului via parametrul 'xml'.");
try {
   $client = new SoapClient('http://127.0.0.1:8080/xml-validator/xml-validator2.asmx?WSDL',
     array('trace' => 1));
   $rez = $client->CheckIfExists ($doc); 
   // apelarea explicita s-ar fi facut prin:
   // $rez = $client->__soapCall('CheckIfExists', array($doc));
   echo 'Cererea efectuata:<pre class="cod">' . 
     htmlspecialchars($client->__getLastRequest(), ENT_QUOTES) . '</pre>';
   echo 'Raspunsul obtinut:<pre class="cod">' . 
     htmlspecialchars($client->__getLastResponse(), ENT_QUOTES) . '</pre>';
   if ($rez == true) {
   	 $rez = $client->Validate ($doc);
   	 echo "<p>Documentul $doc este $rez.</p>";
   } else {
   	 echo "<p>Documentul $doc e inaccesibil (nu exista pe server?)</p>";
   }  
} catch (SOAPFault $exception) {
   echo 'A aparut o exceptie: ' . $exception->faultstring;
}
?>
</body>
</html>